libflux/src/semantic/mod.rs

```diff
@@ -14,7 +14,8 @@ mod sub;
 mod types;
 pub mod walk;
 
-#[allow(warnings)]
+#[cfg(test)]
 mod parser;
-#[allow(warnings)]
+
+#[cfg(test)]
```

Comment:
I saw what you have done here, but I still don't like the `#[cfg(test)]` above those modules 🤔